### ISSUE TITLE
#xtranslate doesn't recognize the Regular match marker (xBase++ dialect)

### ISSUE THREAD

**Issue by Igor-chest** [X-Sharp/XSharpPublic#1077]
Describe the bug
#xtranslate expression is not recognized by the preprocessor (xBase++ dialect).

To Reproduce
```
#xtranslate ASTR(<x>) => ALLTRIM(STR(<x>))

if astr(n := val(fldleft)) == fldleft
endif
```

Expected behavior (xBase ppo)
```
if ALLTRIM(STR(n := val(fldleft))) == fldleft
endif
```

Actual behavior (x# ppo)
```
if astr(n := val(fldleft)) == fldleft
endif
```

I use version 2.12.2.0

**Comment by kpa174** [comment]
Apparently, the problem is with the := operator inside the parentheses.

**Comment by kpa174** [comment]
Maybe it's the same problem.
```
#xtranslate ASSERT(<lOk>,<cMess>[,<cSubSys>]) => if !(<lOk>); ProgramError(<cMess>[,<cSubSys>]); endif
ASSERT(::pos>0,'Incorrect position') // inside any method
```
The colon is probably not recognized.

**Comment by cpyrgas** [comment]
The first reported problem is fixed, but the second one (mentioned here https://github.com/X-Sharp/XSharpPublic/issues/1077#issuecomment-1190001758) is not.

**Comment by RobertvanderHulst** [comment]
> The first reported problem is fixed, but the second one (mentioned here [#1077 (comment)](https://github.com/X-Sharp/XSharpPublic/issues/1077#issuecomment-1190001758)) is not.

Aargh. That is why it is better to limit an issue to only one problem. I had overlooked that Petr had added another problem to this issue.

**Comment by RobertvanderHulst** [comment]
Fixed in 52e7496976239a08ab5cb6ef2d835a553a26933b

**Comment by cpyrgas** [comment]
Confirmed fixed.
